python/ray/air/util/data_batch_conversion.py

```diff
@@ -15,6 +14,13 @@
 except ImportError:
     pyarrow = None
 
+# Lazy import to avoid ray init failures without pandas installed and allow
+# dataset to import modules in this file.
+try:
+    import pandas as pd
+except ImportError:
+    pd = None
```

Comment:
This isn't a lazy import, it's still eager, and will most likely lead to regressions in Ray Core release tests. We should ensure that pandas is not imported at `import ray` time.

Could you try adopting a pattern similar to this? https://github.com/ray-project/ray/blob/d3b8449b821c27cfd39971df9b7b6978bc0dcab5/python/ray/data/_internal/pandas_block.py#L46-L52

Reply:
hmm wait haven't we already merged your PR from 2.1 that removes ray dataset import from critical path ? Thus this only need to ensure it doesn't fail until executing dataset code with pandas

Reply:
discussed via slack, this https://github.com/ray-project/ray/pull/30219 is not merged yet so we got to fall back to dataset internal handling of lazy import. Otherwise perf penalties on dataset imports on critical path of ray.init still apply